### PR TITLE
ARIA-34 Adding get/download resource and render API (as jinja template)

### PR DESCRIPTION
**Pull request by dankilman**

### CODE REVIEW

File: tests/orchestrator/context/test_resource_render.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def test_download_resource_and_render_implicit_ctx_no_variables(tmpdir, ctx):
+    destination = tmpdir.join('destination')
+    ctx.download_resource_and_render(destination=str(destination),
+                                     path=_IMPLICIT_CTX_TEMPLATE_PATH)
+    assert destination.read() == mock.models.DEPLOYMENT_NAME
+
+
+def test_download_resource_and_render_provider_variables(tmpdir, ctx):
```
Comment (ran-z) [inline tests/orchestrator/context/test_resource_render.py]: typo (provided) :sweat_smile:

File: aria/orchestrator/context/common.py [hunk 2]
```diff
@@ -97,19 +99,49 @@ def download_resource(self, destination, path=None):
...
+                                             path=path)
+
+    def download_resource_and_render(self, destination, path=None, variables=None):
+        """
+        Download a blueprint resource from the resource storage render its content as a jinja
+        template using the provided variables. ctx is available to the template without providing it
+        explicitly.
+        """
+        self.download_resource(destination=destination, path=path)
+        with open(destination, 'rb') as f:
```
Comment (ran-z) [inline aria/orchestrator/context/common.py]: why the use of rb and wb? since its rendering doesnt it make sense to use r and w?

Comment (dankilman) [inline aria/orchestrator/context/common.py]: on windows, omitting the `b` will cause newlines to be written as CRLF regardless of how they were originally. So it is definitely required for `wb`. The `rb` is for consistency

Comment (ran-z) [inline aria/orchestrator/context/common.py]: got it thanks

**Pull request merged**